### PR TITLE
chore: Create scaffolding for IPC between UI and background service

### PR DESCRIPTION
**Pull request by gnarea**
This PR defines the API of the IPC-related functions that the UI will use to communicate with the background service.

### CODE REVIEW

File: src/ipc/connectionStatus.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { sleep } from './_utils';
+
+export enum ConnectionStatus {
+  CONNECTED_TO_PUBLIC_GATEWAY,
+  CONNECTED_TO_COURIER,
+  DISCONNECTED_FROM_PUBLIC_GATEWAY,
+  DISCONNECTED_FROM_ALL,
+}
+
+export async function* pollConnectionStatus(): AsyncIterable<ConnectionStatus> {
```
Comment (gnarea) [inline src/ipc/connectionStatus.ts:10]: I believe reactive programming is more common in UI apps, but I have zero experience with it, so I'm using this pattern. Please feel free to replace with a reactive approach.

File: src/ipc/courierSync.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+export class CourierSyncError extends PrivateGatewayError {}
+
+/**
+ * Initialize synchronization with courier and wait until it completes.
+ *
+ * This method will stream the current status of the synchronization.
+ *
+ * @throws CourierSyncError if the synchronization fails at any point
+ */
+export async function* synchronizeWithCourier(): AsyncIterable<CourierSyncStatus> {
```
Comment (gnarea) [inline src/ipc/courierSync.ts:19]: Another case where reactive programming may be better

File: src/ipc/connectionStatus.ts (hunk 1 above)
Comment (liliakai) [reply src/ipc/connectionStatus.ts:10]: No worries. Traditional polling (e.g., while/sleep) is a pain for UI threads but await/async stuff should be easy enough to integrate.

**Review by liliakai — APPROVED** [review]

File: src/ipc/connectionStatus.ts (hunk 1 above)
Comment (gnarea) [reply src/ipc/connectionStatus.ts:10]: Great!

**Pull request merged**